File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        int n=words.length;
        int[] counts=new int[26];
        for(String word: words){
            for(char c: word.toCharArray()){
                counts[c-'a']++;
            }
        }
        for(int val:counts){
            if(val % n!=0){
                return false;
            }
        }
        return true;
    }
}

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.scala
object Solution {
    def makeEqual(words: Array[String]): Boolean = {
        var n=words.length;
        //int[] conts=new int[26];
        var counts:Array[Int]=Array.fill(26)(0);
        for(word <- words){
            for(c <- word.toCharArray()){
                counts(c-'a')+=1;
            }
        }
        for(p <- counts){
            if(p % n!=0){
                return false;
            }
        }
        return true;
    }
}